plugin/backend/ipvariant.d

```diff
@@ -0,0 +1,906 @@
+        }
+    }
+
+   //gmockGlobal(r.classRange, modules.gmock, params);
+    // no singleton in global namespace thus null
+    string[] cifaces;
+    foreach (a; r.namespaceRange()) {
+        eachNs(a, params, modules, null, (USRType usr) => container.find!LocationTag(usr), cifaces);
+    }
+}
+
+import std.container.array;
+@trusted Array!string getDataItems(SUTEnv sut) {
+    Array!string rarr; 
+    foreach(ciface; sut.iface.interfaces) {
+        foreach(ditem; ciface.ditems) {
+                rarr ~= ditem.name;
+        }
+    }
+
+    return rarr;
+}
+
+
+@trusted void generateClass(Generator.Modules inner, string[] ns, string type, SUTEnv sut) {
+    import std.array;
+    import std.string : toLower; 
+    
+    string fqn_ns = ns.join("::"); 
+    string impl_nsname = ns[$-1];
+    
+    with(inner.impl) {        
+        with (class_(impl_nsname ~ "_Impl", "I_" ~ impl_nsname)) {
+            with(private_) {
+                foreach(ciface; sut.iface.interfaces) {
+                    stmt(E(fqn_ns ~ "::" ~ ciface.name ~ "T " ~ toLower(ciface.name)));
+                }
+
+                stmt(E("RandomGenerator* randomGenerator"));
+            }
+            with(public_) {
+
+                with (ctor_body(impl_nsname ~ "_Impl")) { //Generate constructor
+                    stmt(E("randomGenerator") = E("AFL::getRandomGenerator()"));
+                }
+	    
+                with (dtor_body(impl_nsname ~"_Impl")) { //Generate destructor
+             	    sep;
+                }
+
+                with(func_body("void", "Regenerate")) {
+                    stmt(E("fum.V0") = E("randomGenerator->generate(\"Requirer fum V0\")"));
+                    stmt(E("fum.V1") = E("randomGenerator->generate(\"Requirer fum V1\")"));
+                    stmt(E("fum.V2") = E("randomGenerator->generate(\"Requirer fum V2\")"));
+                    stmt(E("fum.V3") = E("randomGenerator->generate(\"Requirer fum V3\")"));
+                    stmt(E("fum.V4") = E("randomGenerator->generate(\"Requirer fum V4\")"));
+                }
+            }
+        }
+    }
+}
+
+void generateCtor(const CppCtor a, Generator.Modules inner) {
+    with (inner.impl.ctor_body(a.name)) {
+    }                        
+}   
+
+void generateDtor(const CppDtor a, Generator.Modules inner) {
+    with (inner.impl.dtor_body(a.name)) {
+    }                        
+}
+
+@trusted void generateCppMeth(const CppMethod a, Generator.Modules inner, string nsname, SUTEnv sut) {
+    import std.string;
+    import std.array;
+    import std.algorithm.searching : canFind;
+    import cpptooling.analyzer.type;
+
+    auto ditems = getDataItems(sut);
+    auto cppm_type = (cast(string)(a.name)).split("_")[0];
+    auto cppm_ditem = (cast(string)(a.name)).split("_")[$-1];  
+
+    if(cppm_type  == "Put") {
+	//Put something in something
+    }
+    else if(cppm_type == "Get" && ditems.array.canFind(cppm_ditem)) {
```

Comment:
Consider what I was mumbling earlier about regarding the data flow.
What is the generator step responsible for?